src/main/java/me/zeroeightsix/kami/feature/module/player/Blink.java

```diff
@@ -4,74 +4,99 @@
 import me.zero.alpine.listener.EventHandler;
 import me.zero.alpine.listener.Listener;
 import me.zeroeightsix.kami.event.PacketEvent;
+import me.zeroeightsix.kami.event.TickEvent;
 import me.zeroeightsix.kami.feature.module.Module;
 import me.zeroeightsix.kami.util.Texts;
 import net.minecraft.client.network.OtherClientPlayerEntity;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.network.Packet;
+import net.minecraft.network.listener.PacketListener;
 import net.minecraft.network.packet.c2s.play.PlayerMoveC2SPacket;
+import net.minecraft.network.packet.s2c.play.PlayerRespawnS2CPacket;
+import net.minecraft.network.packet.s2c.play.PlayerSpawnS2CPacket;
 import net.minecraft.util.Formatting;
 
 import java.util.ArrayDeque;
+import java.util.Objects;
 import java.util.Queue;
 
 @Module.Info(name = "Blink", category = Module.Category.PLAYER)
 public class Blink extends Module {
-    @Setting
-    private boolean packetDiscard = true;
+    @Setting(comment = "Will not attempt to send packets if you are more than 10 blocks away from where you blinked")
+    private boolean isCancelable = true;
     @Setting(comment = "Withhold all packets, not only Movement Packets")
     private boolean withholdAllPackets = true;
-    @Setting
-    private @Setting.Constrain.Range(min = 50d, max = 500d, step = Double.MIN_VALUE) int maxPacketAmount = 100;
+    @Setting(comment = "How many packets are allowed to be cached before they all get canceled")
+    private @Setting.Constrain.Range(min = 20d, max = 200d, step = Double.MIN_VALUE) int maxPacketAmount = 50;
+
+    @Setting(comment = "Send packets gradually instead of all at once")
+    private boolean gradualPacketMode = false;
+    @Setting(comment = "The percent of cached packets to send per tick")
+    private @Setting.Constrain.Range(min = 1d, max = 100d, step = Double.MIN_VALUE) int percentOfPacketsPerTick = 50;
 
     private OtherClientPlayerEntity clonedPlayer; // Fake Player when player blinked
     Queue<Packet> packets = new ArrayDeque<>(); // Create list to hold our packets
+    private float packetsPerClump = 0;
 
     @EventHandler
     public Listener<PacketEvent.Send> listener = new Listener<>(event -> {
         // We don't want to withhold login packets if a player logs out with blink enabled
-        if (event.getPacket() instanceof PlayerMoveC2SPacket || (withholdAllPackets && mc.world != null)) {
+        if (event.getPacket() instanceof PlayerMoveC2SPacket || (withholdAllPackets && mc.world != null && !(event.getPacket() instanceof PlayerRespawnS2CPacket) && !(event.getPacket() instanceof PlayerSpawnS2CPacket) && !(event.getPacket() instanceof PacketListener))) { // maybe use ServerPlayPacketListener
```

Comment:
Uhh, there are no packets that are a descendant of `PacketListener`. I assume you wanted to check for the generic type argument of `Packet` (as a lot of packets implement `Packet<ServerPlayPacketListener>`), but, unfortunately, due to type erasure this is difficult to do.